Source/DafnyServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -154,6 +171,66 @@ public class DafnyModel {
       return result;
     }
 
+    /// <summary> Registered all char values specified by the model </summary>
+    private void RegisterReservedChars() {
+      foreach (var app in fCharToInt.Apps) {
+        int UTFCode = int.Parse(((Model.Integer)app.Result).Numeral);
+        reservedChars.Add(Convert.ToChar(UTFCode));
+      }
+    }
+
+    /// <summary> Registered all int values specified by the model </summary>
+    private void RegisterReservedInts() {
+      reservedNumerals[Type.Int] = new();
+      foreach (var app in fU2Int.Apps) {
+        if (app.Result is Model.Integer integer) {
+          reservedNumerals[Type.Int].Add(integer.AsInt());
+        }
+      }
+    }
+
+    /// <summary> Registered all bool values specified by the model </summary>
+    private void RegisterReservedBools() {
+      foreach (var app in fU2Bool.Apps) {
+        isTrueReserved |= ((Model.Boolean)app.Result).Value;
+      }
+    }
+
+    /// <summary> Registered all real values specified by the model </summary>
+    private void RegisterReservedReals() {
+      reservedNumerals[Type.Real] = new();
+      foreach (var app in fU2Real.Apps) {
+        var valueAsString = app.Result.ToString()?.Split(".")[0] ?? "";
+        if ((app.Result is Model.Real) && int.TryParse(valueAsString, out int value)) {
+          reservedNumerals[Type.Real].Add(value);
+        }
+      }
+    }
+
+    /// <summary> Registered all bv values specified by the model </summary>
+    private void RegisterReservedBitVectors() {
+      foreach (var func in Model.Functions) {
+        if (!Regex.IsMatch(func.Name, "^U_2_bv[0-9]+$")) {
```

Comment:
You may want to [pre-compile](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.-ctor?view=net-6.0#system-text-regularexpressions-regex-ctor(system-string)) this regular expression outside the loop, or even as a static field.

Reply:
I now pre-compile regular expressions here and in other places throughout the counterexample generation code.